### ISSUE TITLE
Sega Classics Arcade Collection (SCD): Black screen when selecting Columns

### ISSUE THREAD

**Issue by CecilFF4** [ekeeke/Genesis-Plus-GX#138]
Using RA 1.6.0, Genesis Plus GX v1.7.4 365a28c. "Sega Classics Arcade Collection 5-in-1" (and presumably 4-in-1, but not sure), loaded as ISO+WAV, freezes when loading Columns. All other games load fine. Music plays fine. When selecting a working game, the game takes a couple seconds at most before starting said game. When selecting Columns, a black screen appears. It is no longer reading the disc at this point. This disc, when loaded in Kega Fusion, works just fine for all games. Log attached.

[log.txt](https://github.com/ekeeke/Genesis-Plus-GX/files/1093746/log.txt)

**Comment by ekeeke** [comment]
Only happens with the Sega CD version, both PAL and NTSC-J version do not have this issue.

**Comment by ekeeke** [comment]
After investigation, the problem is actually caused by unsafe code in the game. The routine that loads then resets Z80 program do not disable interrupts and therefore can be interrupted by VINT callback which itself tries to access Z80 and ends up waiting endlessly for Z80 bus.

It does not happen on real hardware by luck because the routine appears to be delayed and executed after VINT occurence. Similarly, it will or will not happen in emulators depending on the timings they use for CPU instructions, DMA, etc (it's one of the cases where an inaccurate emulator could - again by luck - runs the game just fine).

I fixed it by reducing DMA transfer rate by one slot per line during blanking period, as verified by Mask of Destiny on real hardware (source: http://gendev.spritesmind.net/forum/viewtopic.php?f=22&t=1291&start=34). This was enough to delay the unsafe routine to happen after VINT (like it did with NTSC-J version).

I verified PAL and NTSC-J version still started the game fine after that fix, as well as 4-in-1 version.
